packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2592,9 +2595,16 @@ class RenderPointerListener extends RenderProxyBoxWithHitTestBehavior {
     }
   }
 
+  bool _mouseIsConnected;
   void _handleMouseTrackerChanged() {
-    if (attached)
-      markNeedsPaint();
+    final bool newState = RendererBinding.instance.mouseTracker.mouseIsConnected;
+    if (attached && newState != _mouseIsConnected) {
+      _mouseIsConnected = newState;
```

Comment:
Can you explain why `_mouseIsConnected` doesn't need changing when `attached` is false? Is it possible to be reattached and find `_mouseIsConnected` desync'ed?

Reply:
Oh, yes, I think you're right.  Originally, I was avoiding painting when detached, but avoiding it isn't needed, since it can call `markNeedsPaint` when detached, and that won't do anything until it attaches again.